src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs

```diff
@@ -199,8 +200,8 @@ private static void set_config_items(ChocolateyConfiguration config, ConfigFileS
                 config.CacheLocation = fileSystem.get_temp_path(); // System.Environment.GetEnvironmentVariable("TEMP");
                 // TEMP gets set in EnvironmentSettings, so it may already have 
                 // chocolatey in the path when it installs the next package from
-                // the API. 
-                if(!config.CacheLocation.EndsWith("chocolatey")) {
+                // the API.
+                if(new DirectoryInfo(config.CacheLocation).Name.ToLower() != "chocolatey") {

```

Comment:
Use `fileSystem.get_directory_info()`  and use either `.to_lower()` or `Name.is_equal_to()`. The latter is preferred.